dynamic_programming/iterating_through_submasks.py

```diff
@@ -0,0 +1,46 @@
+'''
+You are given a bitmask m and you want to efficiently iterate through all of
+its submasks. The mask s is submask of m if only bits that were included in
+bitmask are set
+'''
+
+def list_of_submasks(mask)->list:
+
+    """
+    Args:
+        mask : number which shows mask ( always integer > 0, zero does not have any submasks )
+
+    Returns:
+        all_submasks : the list of submasks of mask (mask s is called submask of mask
+        m if only bits that were included in original mask are set
+
+    Raises:
+        AssertionError: mask not positive integer
+
+    >>> list_of_submasks(15)
+    [15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
+    >>> list_of_submasks(13)
+    [13, 12, 9, 8, 5, 4, 1]
+    
```

Comment:
How about a **doctest** where `mask <= 0`

Reply:
Sure I will add it. It should give AssertionError at mask<=0.